jb/itch5/order_book.hpp

```diff
@@ -163,63 +155,44 @@ class order_book {
   bool handle_order_reduced(
       buy_sell_indicator_t side, price4_t px, int reduced_qty) {
     if (side == buy_sell_indicator_t('B')) {
-      return handle_order_reduced(buy_, px, reduced_qty);
+      return buy_.reduce_order(px, reduced_qty);
     }
-    return handle_order_reduced(sell_, px, reduced_qty);
+    return sell_.reduce_order(px, reduced_qty);
   }
 
 private:
-  /**
-   * Refactor handle_add_order()
-   *
-   * @tparam book_side the type used to represent the buy or sell side
-   * of the book
-   * @param side the side of the book to update
-   * @param px the price of the new order
-   * @param qty the quantity of the new order
-   * @returns true if the inside changed
-   */
-  template <typename book_side>
-  bool handle_add_order(book_side& side, price4_t px, int qty) {
-    auto emp_tup = side.emplace(px, 0);
-    emp_tup.first->second += qty;
-    return emp_tup.first == side.begin();
+  typename book_type::buys_t buy_;
+  typename book_type::sells_t sell_;
+};
+
+/**
+ * Configure an order_book config object
+ */
+template <typename book_type>
+class order_book<book_type>::config : public jb::config_object {
```

Comment:
What if you make the `config` object be a nested class of `book_type` instead?  Then you can make the config object be empty for map-based book types, and something different for array-based.  And if we ever invent a new `book_type` we will not need to change this code.